llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -483,12 +488,20 @@ static ScheduleDAGInstrs *createSIMachineScheduler(MachineSchedContext *C) {
   return new SIScheduleDAGMI(C);
 }
 
+static bool getReorderWhileLoadClustering(const MachineFunction *MF) {
+  if (!ReorderWhileLoadClustering)
+    return false;
+  return !!MF->getFunction().getFnAttributeAsParsedInteger(
+      "amdgpu-reorder-loads-while-clustering", 1);
```

Comment:
If you really want to add another one of these explicit boolean attributes, should use getValueAsBool and "true" "false"

Do we really need both mechanisms here? At most I would think the debug cl::opt may be useful for a short period